services/docker/docker-version.service.js

```diff
@@ -0,0 +1,102 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const { anyInteger } = require('../validators')
+const {
+  buildDockerUrl,
+  getDockerHubUser,
+  getMultiPageData,
+} = require('./docker-helpers')
+const { BaseJsonService } = require('..')
+const { NotFound } = require('..')
+
+const buildSchema = Joi.object({
+  count: anyInteger,
+  results: Joi.array()
+    .items(
+      Joi.object({
+        name: Joi.string(),
+        images: Joi.array().items(
+          Joi.object({
+            digest: Joi.string(),
+            architecture: Joi.string(),
```

Comment:
I suspect these should both be required too right?

Reply:
Can't require on the `digest` because it looks like some very old tags somehow may have an `architecture` defined but not a `digest`:

<details><summary>Details</summary>
<p>
```
[server]     {
[server]         "name": "precise-20160624",
[server]         "images": [
[server]             {
[server]                 "digest": "sha256:7bf805cfed25184d896ebe65f2a1558f83f8695214ed5214349693cea6106fe8",
[server]                 "architecture": "amd64"
[server]             }
[server]         ]
[server]     },
[server]     {
[server]         "name": "xenial-20160525",
[server]         "images": [
[server]             {
[server]                 "digest": "sha256:12543ced0f6f754223eff4cca67b56cbf9f384456ab9deaa824ba9ff28bcfdd6",
[server]                 "architecture": "amd64"
[server]             }
[server]         ]
[server]     },
[server]     {
[server]         "name": "wily-20160526",
[server]         "images": [
[server]             {
[server]                 "architecture": "amd64"
[server]             }
[server]         ]
[server]     },
[server]     {
[server]         "name": "trusty-20160526",
[server]         "images": [
[server]             {
[server]                 "architecture": "amd64"
[server]             }
[server]         ]
[server]     },
[server]     {
[server]         "name": "precise-20160526",
[server]         "images": [
[server]             {
[server]                 "architecture": "amd64"
[server]             }
[server]         ]
[server]     },
[server]     {
[server]         "name": "xenial-20160503",
[server]         "images": [
[server]             {
[server]                 "architecture": "amd64"
[server]             }
[server]         ]
[server]     },
[server]     {
[server]         "name": "wily-20160503",
[server]         "images": [
[server]             {
[server]                 "architecture": "amd64"
[server]             }
[server]         ]
[server]     },
[server]     {
[server]         "name": "trusty-20160503.1",
[server]         "images": [
[server]             {
[server]                 "architecture": "amd64"
[server]             }
[server]         ]
[server]     },
[server]     {
[server]         "name": "precise-20160503",
[server]         "images": [
[server]             {
[server]                 "architecture": "amd64"
[server]             }
[server]         ]
[server]     },
[server]     {
[server]         "name": "xenial-20160422",
[server]         "images": []
[server]     },
```
</p>
</details>

I suspect we will be able to change this in future though (eventually).